nucliadb_node_binding/src/writer.rs

```diff
@@ -96,16 +96,23 @@ impl NodeWriter {
 
         let request = NewShardRequest::decode(&mut Cursor::new(metadata)).expect("Error decoding arguments");
         let shard_id = uuid::Uuid::new_v4().to_string();
-        let similarity = VectorSimilarity::try_from(request.similarity).unwrap();
         let metadata = ShardMetadata::new(
             self.shards_path.join(shard_id.clone()),
             shard_id,
-            request.kbid,
-            similarity.into(),
-            Channel::from(request.release_channel),
-            request.normalize_vectors,
+            request.kbid.clone(),
+            Channel::from(request.release_channel()),
         );
-        let new_shard = self.shards.create(metadata);
+        let vector_config = if let Some(req_config) = request.config {
+            VectorConfig::try_from(req_config).map_err(|err| IndexNodeException::new_err(err.to_string()))?
+        } else {
+            #[allow(deprecated)]
+            VectorConfig {
+                similarity: request.similarity().into(),
+                normalize_vectors: request.normalize_vectors,
+                ..Default::default()
+            }
+        };
```

Comment:
You could encapsulate this in the TryFrom implementation if you don't want a repeat yourself (I've seen the same code a couple of times)

Reply:
It's mostly the same but the request is different in each case (NewShard, NewVectorSet, Replication), so I'd have to write a common trait for these and it all gets messy.